### DISCUSSION TITLE
Chats in realtime

### DISCUSSION THREAD

**Discussion by Disregard-Therest** [serverpod/serverpod#1311] [Q&A]
Hi, please, guide me where can I get some info on how to implement the following features:
- unread message count for channels
- notifications about person typing
- last time person was online

By the way, I will be happy to see some kind of an architecture example to implement chats correctly on the flutter side

**Comment by abcnow** [comment]
Here is a video that will teach you how to use web socket with serverpod.. not a chat application but the basic things you should know : how to create tables .. endpoints and UI

https://youtu.be/JhXPFAoWpxU?si=Yy1b8j9-0jJyQnoX

And here is a chat module code that could point you in the right direction 

https://github.com/serverpod/serverpod/tree/main/examples/chat

I hope this helps!
